source/reference/ejson/stringify.txt

```diff
@@ -0,0 +1,506 @@
+
+.. code-block:: javascript
+
+   EJSON.stringify( db.sales.findOne( { custId: 345 } ), null , 5 )
+
+``EJSON.stringify()`` indents each document level five spaces.
+
+.. code-block:: javascript
+
+   {
+        "_id": {
+             "$oid": "64da90c1175f5091debcab26"
+        },
+        "custId": 345,
+        "purchaseDate": {
+             "$date": "2023-07-04T00:00:00Z"
+        },
+        "quantity": 4,
+        "cost": {
+             "$numberDecimal": "100.60"
+        }
+   }
+
+Select Output Fields
+~~~~~~~~~~~~~~~~~~~~
+
+To select a subset of document fields, use an array to set the
+``replace`` option.
+
+.. code-block:: javascript
+
+   EJSON.stringify( db.sales.find().toArray(), [ "quantity", "cost" ] )
+
+EJSON formats the ``quantity`` and ``cost`` for each document.
+
+.. code-block:: javascript
+
+   [{"quantity":4,"cost":{}},{"quantity":3,"cost":{}},{"quantity":9,"cost":{}}]
+
+The ``spacing`` option is not specified in this example, so EJSON
+returns the selected fields on a single line. 
+
+.. _ex-ejson-function:
+
+Use a Function to Transform Output Fields
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To transform field values, use a JavaScript function to set the
+``replace`` option.
+
+.. code-block:: javascript
+
+   let queryResults = db.sales.find().toArray()
+   let replacer = function( key, value ){
+      if ( key === '_id' ) { 
+         value = undefined; 
+      }
+      if ( key === 'quantity' ) { 
+         value = 2 * value; 
+      }
+
+      return value; 
+   }
+
+   EJSON.stringify( queryResults, replacer, 3 )
+
+The function runs recursively against the input object.
+
+.. code-block:: javascript
+
+   [
+      {
+         "custId": 345,
+         "purchaseDate": {
+            "$date": "2023-07-04T00:00:00Z"
+         },
+         "quantity": 8,
+         "cost": {
+            "$numberDecimal": "100.60"
+         }
+      },
+      {
+         "custId": 346,
+         "purchaseDate": {
+            "$date": "2023-07-12T00:00:00Z"
+         },
+         "quantity": 6,
+         "cost": {
+            "$numberDecimal": "175.45"
+         }
+      },
+      {
+         "custId": 486,
+         "purchaseDate": {
+            "$date": "2023-08-01T00:00:00Z"
+         },
+         "quantity": 18,
+         "cost": {
+            "$numberDecimal": "200.53"
+         }
+      }
+   ]
+
+The ``replacer`` function updates two fields, ``_id`` and ``quantity``.
+
+``EJSON.stringify()`` ignores fields with undefined values, so setting
+``_id: undefined`` removes the ``_id`` field from the output string.
+
+The function also modifies the ``quantity`` field in the output string.
+All of the ``quantity`` values are multiplied by two in the output
+string. ``EJSON.stringify()`` does not update the collection.
+
+Write to a File from Inside mongosh
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+To write to a file from within ``mongosh``, call the ``fs`` API. Use
```

Comment:
[question]

Would it be too general to say "use the ``fs`` API here?"

Reply:
Done - I think they both work in this context